File: loops/continue.py
import math


def print_primes(limit):
    for n in range(2, limit):
        if n == 2:
            print(n)
            continue
        if n % 2 == 0:
            continue

        for i in range(3, int(math.sqrt(n)) + 1, 2):
            if n % i == 0:
                break
        else:
            print(n)


print_primes(10000000)
print("This was Python")

File: loops/continue.js
function printPrimes(max) {
  for (let n = 2; n <= max; n++) {
    if (n === 2) {
      console.log(n);
      continue;
    }
    if (n % 2 === 0) {
      continue;
    }

    let isPrime = true;
    for (let i = 3; i <= Math.sqrt(n); i += 2) {
      // fix 2: i <= not 3 <=
      if (n % i === 0) {
        isPrime = false;
        break; // fix 3: break not continue
      }
    }

    if (isPrime) {
      console.log(n); // fix 4: outside the inner loop
    }
  }
}

printPrimes(10000000);
console.log("This was JavaScript");
